File: backend/app/routers/app_releases.py
from fastapi import APIRouter

from app.config import Settings
from app.schemas.app_release import AndroidReleaseManifest

router = APIRouter()


@router.get("/app-releases/android/latest")
def latest_android_release() -> dict[str, object]:
    settings = Settings()
    if settings.android_update_version_code <= 0:
        return {"code": 0, "message": "ok", "data": None}

    manifest = AndroidReleaseManifest(
        version_code=settings.android_update_version_code,
        version_name=settings.android_update_version_name,
        release_notes=settings.android_update_release_notes,
        apk_url=settings.android_update_apk_url,
        apk_sha256=settings.android_update_apk_sha256,
    )

    return {"code": 0, "message": "ok", "data": manifest.model_dump(mode="json")}

File: frontend/src/api/appReleases.ts
import { client, request } from "./client";

export interface AndroidReleaseManifest {
  platform: "android";
  version_code: number;
  version_name: string;
  force_update: true;
  release_notes: string;
  apk_url: string;
  apk_sha256: string;
}

export function getLatestAndroidRelease() {
  return request<AndroidReleaseManifest | null>(client.get("/app-releases/android/latest"));
}
